### PR TITLE
fix: authz exec and coverage directory location

### PR DESCRIPTION
**Pull request by fmorency**
This PR

- Passes environment variable context when creating the AuthZ JSON file containing the nested message to be executed
- Fixes the coverage directory to match `HomeDir()`

Relates #1130 #1158 

@Reecepbcups

### CODE REVIEW

**Review by Reecepbcups — APPROVED** [review]
LGTM

**Comment by fmorency** [comment]
The failing test in the previous run is unrelated to this PR. It also fails on the `v8.5.0` tag.

**Comment by Reecepbcups** [comment]
Failing e2e test is related:

```
    code_coverage_test.go:87: 
        	Error Trace:	/home/runner/work/interchaintest/interchaintest/examples/cosmos/code_coverage_test.go:87
        	            				/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.2.linux-amd64/src/testing/testing.go:1175
        	            				/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.2.linux-amd64/src/testing/testing.go:1353
        	            				/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.2.linux-amd64/src/testing/testing.go:1683
        	Error:      	Should NOT be empty, but was []
        	Test:       	TestCodeCoverage
```

Code coverage is not finding coverage files in the home dir. It was originally pulling `/var/cosmos-chain/manifest-2` which is where it is actually stored (not name), but due to that being the core repo it is possible the output never actually happened for the cov files
image

**Review by Reecepbcups — CHANGES_REQUESTED** [review]
Failing e2e test is related:

```
    code_coverage_test.go:87: 
        	Error Trace:	/home/runner/work/interchaintest/interchaintest/examples/cosmos/code_coverage_test.go:87
        	            				/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.2.linux-amd64/src/testing/testing.go:1175
        	            				/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.2.linux-amd64/src/testing/testing.go:1353
        	            				/home/runner/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.2.linux-amd64/src/testing/testing.go:1683
        	Error:      	Should NOT be empty, but was []
        	Test:       	TestCodeCoverage
```

Code coverage is not finding coverage files in the home dir. It was originally pulling `/var/cosmos-chain/manifest-2` which is where it is actually stored (not name), but due to that being the core repo it is possible the output never actually happened for the cov files
image

File: examples/cosmos/code_coverage_test.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func TestCodeCoverage(t *testing.T) {
 		{
 			Name:          "manifest",
 			Version:       cfgA.Images[0].Version,
-			ChainName:     cfgA.ChainID,
```
Comment (Reecepbcups) [inline examples/cosmos/code_coverage_test.go:55]: ahhhh

**Review by Reecepbcups — APPROVED** [review]

**Pull request merged**